src/CattyTests/ViewController/FormulaEditorSectionViewControllerTests.swift

```diff
@@ -0,0 +1,173 @@
+/**
+ *  Copyright (C) 2010-2021 The Catrobat Team
+ *  (http://developer.catrobat.org/credits)
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+import Nimble
+import XCTest
+
+@testable import Pocket_Code
+
+final class FormulaEditorSectionViewControllerTests: XCTestCase {
+
+    var viewController: FormulaEditorSectionViewController!
+    var formulaEditorViewController: FormulaEditorViewController!
+    var formulaManager: FormulaManager!
+    var spriteObject: SpriteObject!
+
+    override func setUp() {
+        formulaEditorViewController = FormulaEditorViewController()
+
+        formulaManager = FormulaManager(stageSize: Util.screenSize(true), landscapeMode: false)
+        spriteObject = SpriteObject()
+
+    }
+
+    func testInitAndSelectFunctions() {
+        let expectedItems = formulaManager.formulaEditorItemsForFunctionSection(spriteObject: spriteObject)
+        var itemsInTableView = 0
+
+        viewController = FormulaEditorSectionViewController(type: .functions, formulaManager: formulaManager, spriteObject: spriteObject, formulaEditorViewController: formulaEditorViewController)
+        viewController.reloadData()
+
+        XCTAssertTrue(viewController.numberOfSections >= 1)
+
+        for section in 0..<viewController.numberOfSections {
+            for row in 0..<viewController.numberOfRowsInSection[section] {
+
+                let internFormula = InternFormula()
+                formulaEditorViewController.internFormula = internFormula
+                XCTAssertEqual(0, internFormula.getInternTokenList()?.count)
+
+                viewController.tableView(viewController.tableView, didSelectRowAt: IndexPath(row: row, section: section))
+
+                assertThatInputIsValid(for: internFormula, having: expectedItems)
+                itemsInTableView += 1
+            }
+        }
+
+        XCTAssertEqual(expectedItems.count, itemsInTableView)
+    }
+
+    func testInitAndSelectLogicSection() {
```

Comment:
Thanks! Do make the tests complete, please add one UI test each for selecting a logic item, a function, an object property and a sensor. You can leave out the data section, since this is already tested thoroughly.

Reply:
Hi Michael, I have pushed a new commit with these UI tests. Please review it.